backend/clubs/views.py

```diff
@@ -2705,6 +2707,26 @@ def get_serializer_class(self):
         return UserClubVisitSerializer
 
 
+class SearchQueryViewSet(viewsets.ModelViewSet):
+    """
+    create: Add a new search query
+
+    list: Superuser sees all queries, any other user sees only their own
+    """
+
+    permission_classes = [IsAuthenticated]
+    http_method_names = ["get", "post"]
+
+    def get_queryset(self):
+        if self.request.user.is_superuser:
+            return SearchQuery.objects.all()
+        else:
+            return SearchQuery.objects.filter(person=self.request.user)
+
+    def get_serializer_class(self):
+        return SearchQuerySerializer
```

Comment:
You can do `serializer_class = ` instead of this, which is recommended if you're just returning the serializer object.